doc/source/glossary.rst

```diff
@@ -54,6 +145,84 @@ Glossary
        Any sequence that can be interpreted as an ndarray.  This includes
        nested lists, tuples, scalars and existing arrays.
 
+   array scalar
+      For uniformity in handling operands, NumPy treats
+      a :doc:`scalar <reference/arrays.scalars>` as an array of zero
+      dimension.
+
+
+   attribute
+       A property of an object that can be accessed using ``obj.attribute``,
+       e.g., ``shape`` is an attribute of an array::
+
+         >>> x = np.array([1, 2, 3])
+         >>> x.shape
+         (3,)
+
+   axis
+
+      Another term for an array dimension. Axes are numbered left to right;
+      axis 0 is the first element in the shape tuple.
+
+      In a two-dimensional vector, the elements of axis 0 are rows and the
+      elements of axis 1 are columns.
+
+      In higher dimensions the picture changes. NumPy prints
+      higher-dimensional vectors as replications of row-by-column building
+      blocks, as in this three-dimensional vector:
+
+          >>> a = np.arange(12).reshape(2,2,3)
+          >>> a
+          array([[[ 0,  1,  2],
+                  [ 3,  4,  5]],
+          <BLANKLINE>
+                 [[ 6,  7,  8],
+                  [ 9, 10, 11]]])
+
+      ``a`` is depicted as a two-element array whose elements are 2x3 vectors.
+      From this point of view, rows and columns are the final two axes,
+      respectively, in any shape.
+
+      This rule helps you anticipate how a vector will be printed, and
+      conversely how to find the index of any of the printed elements. For
+      instance, in the example, the last two values of 8's index must be 0 and
+      2. Since 8 appears in the second of the two 2x3's, the first index must
+      be 1:
+
+        >>> a[1,0,2]
+        8
+
+      A convenient way to count dimensions in a printed vector is to
+      count ``[`` symbols after the open-parenthesis. This is
+      useful in distinguishing, say, a (1,2,3) shape from a (2,3) shape:
+
+        >>> a = np.arange(6).reshape(2,3)
+        >>> a.shape
+        (2, 3)
+        >>> a
+        array([[0, 1, 2],
+               [3, 4, 5]])
+
+        >>> a = np.arange(6).reshape(1,2,3)
+        >>> a.shape
+        (1, 2, 3)
+        >>> a
+        array([[[0, 1, 2],
+                [3, 4, 5]]])
+
+
+   .base
+
+      If an array does not own its memory, then its
+      :doc:`base <reference/generated/numpy.ndarray.base>` attribute
+      returns the object whose memory the array is referencing. That object
+      may be borrowing the memory from still another object, so the
+      owning object may be ``a.base.base.base...``. Despite advice to the
+      contrary, testing ``base`` is not a surefire way to determine if two
+      arrays are :term:`view`\ s.
```

Comment:
This feels like it belongs in the docs for `ndarray.base` instead of the glossary

Reply:
@bjnath?

Reply:
I glossed it in hope of reaching newcomers who might not think to look up ndarray.base. 

At the same time, the docstring would benefit from a warning that one call to base may not be the end of the line. 

Perhaps these approaches can coexist.

Reply:
Perhaps then just
```
   base
       The `ndarray.base` property of arrays.
```

Reply:
> I glossed it in hope of reaching newcomers who might not think to look up ndarray.base.

I'm not convinced newcomers will even have heard of `base`. Ideally we'd just make sure to link to `ndarray.base` the first time we mention it. I don't mind having glossary entries that link to methods and properties to start users on the right track though.

Reply:
They serve different purposes. The level of explanation in the glossary entry is inappropriate for a reference entry. We don't  want docstrings to include an essay for newbies. The docstring should get just one additional sentence.  But +1 for a link.

Reply:
> I'm not convinced newcomers will even have heard of base.

There's advice on the web that base is how you can tell a copy from a view. That's the only reason I troubled to make an entry.